### ISSUE TITLE
regression: x axis range is extended multiple times when plotting multiple data sets

### ISSUE THREAD

**Issue by fwyzard** [mwaskom/seaborn#1657]
Moving from version 0.8.1 to 0.9.0 there is a regression in the x axis range produced by `lmplot` when plotting multiple data sets (e.g. multiple `hue`s).

For example, with 0.8.1 I get
![b3f89ee](https://user-images.githubusercontent.com/4069793/51623202-78fd3300-1f38-11e9-8cfb-0b15aa4b9e71.png)

while with 0.9.0 I get
![0b7f68f](https://user-images.githubusercontent.com/4069793/51623219-81ee0480-1f38-11e9-92c7-513b38f5d19b.png)

I bisected the history and found that the incorrect behaviour is introduced by 0b7f68f .

Indeed, here is the plot I get reverting 0b7f68f on top of v0.9.0:
![plot](https://user-images.githubusercontent.com/4069793/51623335-bfeb2880-1f38-11e9-8117-1254badd77ec.png)

**Comment by fwyzard** [comment]
The test I used is
```python
import random
import pandas as pd
import seaborn as sns

df = pd.DataFrame()
size = 1000
df['c'] = [ random.sample(range(0, 10), 1)[0] for i in range(size) ]
df['x'] = [ random.random() * 100. for i in range(size) ]
df['y'] = [ df['x'][i] + df['c'][i] * 5. - 10. + random.random() * 10. for i in range(size) ]

plot = sns.lmplot(data=df, x='x', y='y', hue='c', ci=None)
plot.savefig('plot.png')
```

**Comment by mwaskom** [comment]
This works as expected with #1896

**Comment by fwyzard** [comment]
Mhm, no, it doesn't.

Using the HEAD of the `master` branch,  at 
  * b47d84383b4c Avoid floating point error with maximum husl sat/lum (#2000)

and the same test as before
```python
import random
import pandas as pd
import seaborn as sns

df = pd.DataFrame()
size = 1000
df['c'] = [ random.sample(range(0, 10), 1)[0] for i in range(size) ]
df['x'] = [ random.random() * 100. for i in range(size) ]
df['y'] = [ df['x'][i] + df['c'][i] * 5. - 10. + random.random() * 10. for i in range(size) ]

plot = sns.lmplot(data=df, x='x', y='y', hue='c', ci=None)
plot.savefig('plot.png')
```
the resulting plot is
![plot](https://user-images.githubusercontent.com/4069793/76928651-83b81b80-68e2-11ea-8934-f3cf5b973002.png)

**Comment by mwaskom** [comment]
@fwyzard I don't replicate. Using exactly your code, I get:

![image](https://user-images.githubusercontent.com/315810/76963786-fbdd0c00-68f7-11ea-917c-b8a76ca8c124.png)

I don't think your Python session is actually using the right version of seaborn because, as part of the fix in #1896, the `truncate` default was changed to `True`.

See for yourself: https://github.com/mwaskom/seaborn/blob/master/seaborn/regression.py#L565

What does `seaborn.__version__` say in the session where you are seeing that plot?

**Comment by fwyzard** [comment]
Ah, I see.

For whatever reason python2 still picks up an old version of seaborn (`0.9.0`),  while python3 uses the local one (``0.11.0.dev0`), with the correct plot.

Thanks for the fix and sorry for the noise.
